src/libraries/Common/src/System/Reflection/Metadata/TypeNameParserHelpers.cs

```diff
@@ -0,0 +1,747 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+using static System.Array;
+using static System.Char;
+using static System.Int32;
+
+#nullable enable
+
+namespace System.Reflection.Metadata
+{
+    internal static class TypeNameParserHelpers
+    {
+        internal const int SZArray = -1;
+        internal const int Pointer = -2;
+        internal const int ByRef = -3;
+        private const char EscapeCharacter = '\\';
+#if NET8_0_OR_GREATER
+        private static readonly SearchValues<char> _endOfTypeNameDelimitersSearchValues = SearchValues.Create(".+");
```

Comment:
This is ok but FWIW it won't actually improve anything over just using LastIndexOfAny('.', '+') at the call site.

Reply:
Thanks for pointing this out. I am going to remove it as it's going to reduce the number of `#if`defs

Reply:
(Just to be clear, I was referring only to the first one. The second one is useful.)